warehouse/query-core/src/main/java/datawave/query/jexl/DatawavePartialInterpreter.java

```diff
@@ -0,0 +1,887 @@
+                if (allIdentifiers && !(current instanceof ASTIdentifier)) {
+                    allIdentifiers = false;
+                }
+            }
+        }
+        
+        // If all ASTIdentifiers, then traverse the whole queue. Otherwise, we can attempt to short circuit.
+        State result = null;
+        if (allIdentifiers) {
+            // Likely within a function and must visit every child in the stack.
+            // Failure to do so will short circuit value aggregation leading to incorrect function evaluation.
+            while (!children.isEmpty()) {
+                // Child nodes were put onto the stack left to right. PollLast to evaluate left to right.
+                result = (State) interpretOr(children.pollLast().jjtAccept(this, data), result);
+            }
+        } else {
+            
+            // We are likely within a normal union and can short circuit
+            while (!children.isEmpty()) {
+                // Child nodes were put onto the stack left to right. PollLast to evaluate left to right.
+                result = (State) interpretOr(children.pollLast().jjtAccept(this, data), result);
+                
+                if (result.matched && !result.fromUnknown) {
+                    break;
+                }
+            }
+            return result;
+        }
+        return result;
+    }
+    
+    /**
+     * Interpret the result of a union via pairwise element evaluation
+     *
+     * @param left
+     *            the current element
+     * @param right
+     *            the current evaluation state of this union
+     * @return the updated evaluation state of this union
+     */
+    @Override
+    public Object interpretOr(Object left, Object right) {
+        State leftState = null;
+        State rightState = null;
+        
+        if (left != null) {
+            if (left instanceof State) {
+                leftState = (State) left;
+            } else if (left instanceof FunctionalSet) {
+                leftState = new State(true, false, (FunctionalSet) left);
+            }
+        }
+        
+        if (right != null) {
+            if (right instanceof State) {
+                rightState = (State) right;
+            } else if (right instanceof FunctionalSet) {
+                rightState = new State(true, false, (FunctionalSet) right);
+            }
+        }
+        
+        if (leftState != null && rightState != null) {
+            leftState.mergeOr(rightState);
+            return leftState;
+        } else if (leftState != null) {
+            return leftState;
+        } else if (rightState != null) {
+            return rightState;
+        } else {
+            // if we got here then we are likely dealing with a function that takes a union of null fields as arguments
+            // for example, filter:notNull(NULL1 || NULL2). return false with an initialized empty functional set
+            return new State(true);
```

Comment:
This would require both states to be null, but null or null is surely null?

Reply:
This code is a bit old, it returns a false state.